### PR TITLE
Notebook-MCTS-material-advantage

### PR DESCRIPTION
**Pull request by NOVAdrien**
## What does this PR do?

Key insights about the PR.

## Linked Issues

- Closes #?
- #?

## Summary by Sourcery

Introduce a material-based heuristic for MCTS and extend the example notebook with sections demonstrating its use alongside neural network models.

New Features:
- Add MaterialHeuristic class to compute position value by material advantage for MCTS
- Include notebook section testing the material-advantage heuristic and selecting the best move with MCTS

Enhancements:
- Update self-play MCTS NN notebook imports, execution counts, and sampling parameters for quicker demonstrations
- Reduce model simulation count and game move limit in the notebook for faster execution
- Reorganize search.py imports and introduce PIECE_VALUES constant

Documentation:
- Extend selfplay-mcts-nn notebook with headings, import cleanup, and commented templates for heuristic-based self-play
- Add example of rendering the MCTS search tree (with graphviz requirement noted)

### CODE REVIEW

File: src/lczerolens/search.py [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
-from lczerolens.board import LczeroBoard
 import numpy as np
-from tensordict import TensorDict
 import torch
+from lczerolens.board import LczeroBoard
 from lczerolens.model import ForceValue, LczeroModel
+from tensordict import TensorDict
 from typing import Dict, Protocol, Tuple, Optional
 
+PIECE_VALUES = {
```
[the remark this answers is not in the document]
Comment (NOVAdrien) [reply src/lczerolens/search.py:11]: OOps, oublié d'enlever

File: src/lczerolens/search.py [hunk 2]
```diff
@@ -55,6 +64,40 @@ def evaluate(
...
+        chess.BISHOP: 3,
+        chess.ROOK: 5,
+        chess.QUEEN: 9,
+        chess.KING: 0,
+    }
+
+    def evaluate(
+        self,
+        board: LczeroBoard,
+        piece_values: Optional[Dict[int, int]] = None,
```
[the remark this answers is not in the document]
Comment (NOVAdrien) [reply src/lczerolens/search.py:82]: Right

File: src/lczerolens/search.py [hunk 3]
```diff
@@ -55,6 +64,40 @@ def evaluate(
...
+        """
+        if piece_values is None:
+            piece_values = self.piece_values
+        us, them = board.turn, not board.turn
+        relative_value = 0
+        for piece in range(1, 7):
+            relative_value += len(board.pieces(piece, us)) * piece_values[piece]
+            relative_value -= len(board.pieces(piece, them)) * piece_values[piece]
+
+        value = torch.tensor([relative_value / 39.0], dtype=torch.float32)
```
[the remark this answers is not in the document]
Comment (NOVAdrien) [reply src/lczerolens/search.py:95]: Okay

File: tests/unit/tests_search.py [hunk 4]
```diff
@@ -63,6 +62,56 @@ def test_dummy_heuristic_output():
...
+    ("w", "q", -9.0),
+    ("b", "q", 9.0),
+    ("w", "K", 100.0),
+    ("b", "K", -100.0),
+    ("w", "k", -100.0),
+    ("b", "k", 100.0),
+]
+
+
+@pytest.mark.parametrize("turn,piece,expected_value", test_cases)
```
[the remark this answers is not in the document]
Comment (NOVAdrien) [reply tests/unit/tests_search.py:93]: Ok, I parametrized based on the turn ("w", "b") and the pieces ("P", "p", "K", "k", "B", "b", "R", "r", "Q", "q", "K", "k"), so there are 24 tests in total.
